### PR TITLE
Add crypto stream and rest endpoints

### PR DESCRIPTION
**Pull request by noramehesz**

### CODE REVIEW

File: lib/resources/datav2/crypto_websocket_v2.js [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+    this.on(EVENT.STOCK_QUOTES, (quote) => fn(quote));
+  }
+
+  onStockBar(fn) {
+    this.on(EVENT.STOCK_BARS, (bar) => fn(bar));
+  }
+
+  onStockDailyBar(fn) {
+    this.on(EVENT.STOCK_DAILY_BARS, (dailyBar) => fn(dailyBar));
+  }
```
Comment (gnvk) [inline lib/resources/datav2/crypto_websocket_v2.js:232]: These should be Crypto, right?

File: lib/resources/datav2/rest_v2.js [hunk 2]
```diff
@@ -156,6 +152,123 @@ async function getSnapshots(symbols, config) {
...
 }
 
+async function* getCryptoTrades(
+  symbol,
+  options = { start: null, end: null, limit: null, exchanges: null },
+  config
+) {
+  if (options.exchanges == null) {
+    throw new Error("exchange is required");
+  }
```
Comment (gnvk) [inline lib/resources/datav2/rest_v2.js:162]: For the historical endpoints exchanges is not required.

File: lib/resources/datav2/crypto_websocket_v2.js [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+    });
+  }
+
+  reconnecting() {
+    setTimeout(() => {
+      if (this.session.backoff) {
+        this.session.reconnectTimeout += this.session.backoffIncrement;
+        if (this.session.reconnectTimeout > this.session.maxReconnectTimeout) {
+          this.session.reconnectTimeout = this.session.maxReconnectTimeout;
+        }
```
Comment (SC4RECOIN) [inline lib/resources/datav2/crypto_websocket_v2.js:102]:
```suggestion
        const { reconnectTimeout, backoffIncrement, maxReconnectTimeout } = this.session
        this.session.reconnectTimeout = Math.min(backoffIncrement + reconnectTimeout, maxReconnectTimeout)
```

File: lib/resources/datav2/crypto_websocket_v2.js [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+      bars.length > 0 ||
+      dailyBars.length > 0
+    ) {
+      const msg = {
+        action: "subscribe",
+        trades: this.session.subscriptions.trades,
+        quotes: this.session.subscriptions.quotes,
+        bars: this.session.subscriptions.bars,
+        dailyBars: this.session.subscriptions.dailyBars,
+      };
```
Comment (SC4RECOIN) [inline lib/resources/datav2/crypto_websocket_v2.js:154]: you already extracted these values on ln 141
```js
const msg = {
    action: "subscribe", trades, quotes, bars, dailyBars
};
```
or you could just do this
```js
const msg = { action: "subscribe", ...this.session.subscriptions };
```

**Review by gnvk — APPROVED** [review]
LGTM

**Pull request merged**